src/tools/auth0/handlers/actions.ts

```diff
@@ -203,11 +204,33 @@ export default class ActionHandler extends DefaultAPIHandler {
     const changes = await this.calcChanges(assets);
 
     await super.processChanges(assets, changes);
+
+    const postProcessedActions = await (async () => {
+      this.existing = null; //Clear the cache
+      const actions = await this.getType();
+      return actions;
+    })();
+
     // Deploy actions
     const deployActions = [
-      ...changes.create.filter((action) => action.deployed),
+      ...changes.create
+        .filter((action) => action.deployed)
+        .map((actionWithoutId) => {
+          // Add IDs to just-created actions
+          const actionId = postProcessedActions?.find((postProcessedAction) => {
+            return postProcessedAction.name === actionWithoutId.name;
+          })?.id;
+
+          const actionWithId = {
+            ...actionWithoutId,
+            id: actionId,
+          };
+          return actionWithId;
+        })
+        .filter((action) => !!action.id),
```

Comment:
This is the bulk of this PR. Immediately after processing changes through `super.processChanges(...)`, re-fetch the actions to receive the newly-created IDs and match those up with with the existing actions objects for deployment.